### PR TITLE
[Feature] Benefit plan download UI

### PR DESCRIPTION
**Pull request by JavaLavaMT**
image

image

image

image

### CODE REVIEW

File: app/components/benefits/plans/ae.hbs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  <span class="ui right floated">
+    <div class="ui image carrier_logo">
+      <img src={{this.carrier.logo}}>
+    </div>
+  </span>
+  You must request a carrier authorization code and company id from the carrier. Please call {{this.carrier.phone}} and
+  ask for
+  a Granite HR authorization code. The carrier will request some identifying information to verify your account and
+  then
+  provide an authorization code and company id. Do not enter or share this information anywhere except for this page.
```
Comment (James1x0) [inline app/components/benefits/plans/ae.hbs:11]:
```suggestion
  You must request a carrier authorization code and company id from the carrier. Please call {{this.carrier.phone}} and ask for a Granite HR authorization code. The carrier will request some identifying information to verify your account and then provide an authorization code and company id. Do not enter or share this information anywhere except for this page.
```

File: app/components/benefits/plans/ae.hbs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+</p>
+<div class="field">
+  <Input::UiInput
+    @type="text"
+    @inputId="auth-code"
+    @keyup={{fn this.updateCarrierData 'carrierAuthCode'}}
+    @placeholder="Code provided by the carrier" />
```
Comment (James1x0) [inline app/components/benefits/plans/ae.hbs:18]:
```suggestion
    @placeholder="Authorization code provided by the carrier" />
```

File: app/config.js [hunk 3]
```diff
@@ -64,41 +64,13 @@ const notifyDefaults = { autoClear: true };
...
-  },
-    {
-  name: 'Ho ho ho ho',
-  key: 'ae',
-  logo: "aetrustlogo.png", 
-  phone: '1 (800) 123-1234'
+    name:  'Associated Employers Group Benefit Plan & Trust',
+    key:   'ae',
+    logo:  'https://granitehr-dev.s3.amazonaws.com/uploads/5d6e7d2fd41c6c27661eb678/5f08c43009c42358fdc6fc3e_83SDcMJFX8szueppRoV3.png',
+    phone: '1 (800) 123-1234'
```
Comment (James1x0) [inline app/config.js:70]:
```suggestion
    phone: '1 (406) 248-6224'
```

File: app/controllers/account/benefits/carrier-link.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  @service data
+  @service ajax
+
+  get carrierRoute () {
+    return `benefits/plans/${this.model.key}`;
+  }
+
+  @action
+  async linkCarrier () {
+    console.log('jajsjas', this);
```
Comment (James1x0) [inline app/controllers/account/benefits/carrier-link.js:15]:
```suggestion
```

File: app/controllers/account/benefits/carrier-link.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    const { success, error } = this.data.createStatus('carrierLink');
+    try {
+      await this.ajax.post('/api/v1/benefits/plan-download', {
+        data: {
+          carrier:     this.model.key,
+          carrierData: this.carrierData
+        }
+      });
+      success('Successfully downloaded plan data.');
```
Comment (James1x0) [inline app/controllers/account/benefits/carrier-link.js:24]: Transition us

File: app/controllers/account/benefits/carrier-link.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class CarrierLinkController extends Controller {
+  @service data
+  @service ajax
+
+  get carrierRoute () {
+    return `benefits/plans/${this.model.key}`;
+  }
```
Comment (James1x0) [inline app/controllers/account/benefits/carrier-link.js:11]: Not sure this is used

File: app/templates/account/benefits/index.hbs [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <div class="ui relaxed divided list">
   <h2 class="ui center aligned header">Nothing to see here.</h2>
   <h5 class="center aligned text header">There are no benefits set up. To offer benefit administration through Granite,
-    <LinkTo @route="account">add a carrier</LinkTo> or <LinkTo @route="account">manually add plans</LinkTo>.
+    <LinkTo @route="account.benefits.plans">add a carrier</LinkTo> or <LinkTo @route="account">manually add plans</LinkTo>.
```
Comment (James1x0) [inline app/templates/account/benefits/index.hbs:4]: Can we move `account.benefits.plans` to `account.benefits.plans.integrations.index` and then move `account.benefits.carrier-link` to `account.benefits.plans.integrations.carrier-link`

File: app/controllers/account/benefits/carrier-link.js (hunk 6 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/carrier-link.js:11]: https://github.com/associatedemployers/granite/blob/add-carrier-link-page/app/templates/account/benefits/carrier-link.hbs#L8

File: app/components/benefits/plans/ae.hbs (hunk 1 above)
Comment (JavaLavaMT) [reply app/components/benefits/plans/ae.hbs:11]: 👍

File: app/components/benefits/plans/ae.hbs (hunk 2 above)
Comment (JavaLavaMT) [reply app/components/benefits/plans/ae.hbs:18]: 🏁

File: app/config.js (hunk 3 above)
Comment (JavaLavaMT) [reply app/config.js:70]: 🏁

File: app/controllers/account/benefits/carrier-link.js (hunk 5 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/carrier-link.js:24]: 🏁

File: app/templates/account/benefits/index.hbs (hunk 7 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/index.hbs]: 🏁

File: app/controllers/account/benefits/carrier-link.js (hunk 4 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/carrier-link.js]: 🏁

**Pull request merged**